src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs

```diff
@@ -20,27 +21,77 @@ public override DocumentIRNode ExecuteCore(RazorCodeDocument codeDocument, Docum
             var parserOptions = syntaxTree.Options;
 
             var designTime = parserOptions.DesignTimeMode;
-            var walker = new DirectiveWalker(designTime);
+            var walker = new DirectiveWalker();
             walker.VisitDocument(irDocument);
 
+            var classNode = walker.ClassNode;
+            foreach (var node in walker.FunctionsDirectiveNodes)
+            {
+                node.Parent.Children.Remove(node);
```

Comment:
Eventually should not remove all these nodes here and let the optimization pass clean this up.